### DISCUSSION TITLE
[@types/xmldom] parseFromString method of DOMParser class can return undefined

### DISCUSSION THREAD

**Discussion by arcogabbo** [DefinitelyTyped/DefinitelyTyped#65454] [Issues with a @types package] [answered]
Currently, `@types/xmldom@0.1.31` doesn't support this([source](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bd9e8f231b9a0a3cadc629cff53c9d878ea371d0/types/xmldom/index.d.ts#LL27C20-L27C20)).

Reproducible script:
```typescript
import * as xmldom from "xmldom";

const p=new xmldom.DOMParser();

const value=p.parseFromString("")

console.log(value.childNodes)
```

[a comment by an automated account is not shown]

**Comment by karfau** [comment, marked as answer]
Hey there, I'm the current maintainer of the xmldom library and there have been some changes that you might not be aware of:
- the project was forked into https://github.com/xmldom/xmldom
- we were forced to publish to the new scoped package name `@xmldom/xmldom` on npm starting from version 0.7.0
- all versions of the `xmldom` package (up to 0.6.0) have security vulnerabilities, so it is advisable to update to the new package name
- since version 0.7.1 the package ships with type definitiions
- the `latest` version of the package is 0.8.7
- a lot of time has already been invested (and has to be invested, no ETA) into version 0.9.0, which is currently available as a beta version with the `next` tag, for which I'm planning to throw an error when no `Document` is returned. (This is not spec compliant, but I think it is still better than returning `undefined`.) Currently that next version provides a type definition that indicate that it might return `undefined`, which is going to change before 0.9.0 is released as `latest`.

With all that being said, I guess the `@types/xmldom` package could be marked as deprecated, in the same way that all versions of the `xmldom` package (beside 0.6.0 because we no longer are able to do that) already are marked as deprecated.
